### PR TITLE
Eip4844 state upgrade

### LINKED ISSUE #6454: Implement Eip4844StateUpgrade

### PR DESCRIPTION
**Pull request by tbenr**
- Makes 4844's SpecLogic using 4844 schema definition
- fixes Capella state upgrade too

fixes #6454

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+              state.setFork(
+                  new Fork(
+                      preState.getFork().getCurrentVersion(),
+                      specConfig.getEip4844ForkVersion(),
+                      epoch));
+
+              state.setLatestExecutionPayloadHeader(
+                  schemaDefinitions.getExecutionPayloadHeaderSchema().getDefault());
+              state.setNextWithdrawalValidatorIndex(UInt64.ZERO);
```
Comment (zilm13) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java]: I think there should be Capella's pre `next_withdrawal_validator_index`

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+              state.setNextSyncCommittee(preStateCapella.getNextSyncCommittee());
+              state.setInactivityScores(preStateCapella.getInactivityScores());
+
+              state.setFork(
+                  new Fork(
+                      preState.getFork().getCurrentVersion(),
+                      specConfig.getEip4844ForkVersion(),
+                      epoch));
+
+              state.setLatestExecutionPayloadHeader(
```
Comment (zilm13) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java]: I think we should copy Capella's pre LatestExecutionPayloadHeader contents here

**Review by zilm13 — COMMENTED** [review]
I think we should keep these fields from Capella

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java (hunk 2 above)
Comment (zilm13) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java]: btw same issue with our Capella upgrade:
https://github.com/ethereum/consensus-specs/blame/dev/specs/capella/fork.md#L73

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java (hunk 1 above)
Comment (tbenr) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java]: you're absolutely right

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java (hunk 2 above)
Comment (tbenr) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java]: yeah.. capella is buggy here to i think, but execution header changes in capella. Need to look into the spec WRT state upgrade

Comment (tbenr) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/eip4844/forktransition/Eip4844StateUpgrade.java]: https://github.com/ethereum/consensus-specs/blob/dev/specs/capella/fork.md
going to fix in capella and here

**Review by zilm13 — APPROVED** [review]
LGTM

**Pull request merged**